### ISSUE TITLE
resolveData breaks when data has null values

### ISSUE THREAD

**Issue by afjusto** [eclipsesource/jsonforms#1607]
**Describe the bug**
The `resolveData` helper breaks when the `data` has properties (of type object) with `null` values.

**To Reproduce**
1. Define a schema with a `object` property (`personalData`):
```js
const schema = {
    type: "object",
    properties: {
      personalData: {
        type: "object",
        properties: {
          age: {
            type: "integer"
          }
        }
      }
    }
  };
```
2. Define a uischema with a control which references a nested property (`age`) of the `personalData` object:
```js
  const uischema = {
    type: "VerticalLayout",
    elements: [
      {
        type: "Control",
        scope: "#/properties/personalData/properties/age"
      }
    ]
  };
```
3. Set the data in which the `personalData` object property is `null`:
```js
  const data = {
    personalData: null
  };
```
4. The following error is thrown:
```
Cannot read property 'hasOwnProperty' of null
    at eval (https://6lw20.csb.app/node_modules/
jsonforms/core/lib/util/resolvers.js:53:26
    at Object.exports.resolveData [as data] (https://6lw20.csb.app/node_modules/
jsonforms/core/lib/util/resolvers.js:51:10
```

You can check [this codesandbox](https://codesandbox.io/s/vigorous-borg-6lw20) with the full reproduction case.

**Expected behavior**
I'd expect the `resolveData` to properly handle `null` values (before blindly calling `hasOwnProperty)`, and return `undefined` in such cases.

I believe that updating the logic in https://github.com/eclipsesource/jsonforms/blob/master/packages/core/src/util/resolvers.ts#L60 to something like:
```js
if (
       !curInstance || !curInstance.hasOwnProperty(decodedSegment)
) {
        return undefined;
  }
``` 
would prevent this issue.

**Browser**
 - Browser: Chrome
 - Version: 83.0.4103.106

**Used Setup (please complete the following information):**
 - Framework: react + redux in my app (or just react in the codesandbox)
 - RendererSet: custom set in my app (or material in the codesandbox)

**Additional context**

I've been using an utility function in my application to remove `null` values from the `data` payload sent by the API, but this workaround became an issue since it's sometimes helpful to have `null`s in the `data` (e.g. to apply visibility rules when a given property is `null`).

**Comment by sdirix** [comment]
Hi @afjusto, good find! Thanks for the analysis and extensive write-up. 

We should definitely harden JSON Forms against any unexpected types in the data. Besides the suggested change we should also check the update logic whether it will sensibly replace `null` or any other invalid value like for example `true`.

We'll take a closer look soon. If you like you could of course also contribute :wink:

**Comment by sdirix** [comment]
@afjusto The fix is part of the already released `2.4.1-alpha.3` if you'd like to consume and/or test it. Thanks again for the report ;)

**Comment by afjusto** [comment]
Hi @sdirix, thanks for the heads up.

Just tested it and it's working as expected :+1: will wait for the next stable release to use it.
